src/containers/Header/PreviewFlag.js

```diff
@@ -1,13 +1,33 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useTranslation } from "react-i18next";
+import { useSelector } from 'react-redux';
 
 export default function PreviewFlag({ name }) {
-    if(!name) {
-        return null
+    const flagName = name?.toLowerCase();
+    const { t } = useTranslation();
+    const theme = useSelector(state=>state?.theme);
+
+    useEffect(() => {
```

Comment:
@queuing4oranges I think this is a bit overkill, lets remove the whole `useeffect` and instead, you can do conditional chaining directly in `<span>` as per this example

```
<span className={`badge badge-warning mr-4 ${theme == "light" ? "preview-flag-light" : "preview-flag-dark"}`>
```

Reply:
cool! thanks. looks a lot better :smile: